examples/Remarketing/AddMerchantCenterDynamicRemarketingCampaign.php

```diff
@@ -0,0 +1,454 @@
+            // Optional: Call to action text.
+            // Valid texts: https://support.google.com/adwords/answer/7005917
+            'call_to_action_text' => new StringValue(['value' => 'Apply Now']),
+            // Optional: Creates a logo image and set it to the ad.
+            'logo_images' => [
+                new AdImageAsset([
+                    'asset' => new StringValue(['value' => $logoImageResourceName])
+                ])
+            ],
+            // Optional: Creates a square logo image and set it to the ad.
+            'square_logo_images' => [
+                new AdImageAsset([
+                    'asset' => new StringValue(['value' => $logoImageResourceName])
+                ])
+            ]
+            // Whitelisted accounts only: Sets color settings using hexadecimal values.
+            // Sets allowFlexibleColor to false if you want your ads to render by always
+            // using your colors strictly.
+            // 'main_color' => new StringValue(['value' => '#0000ff']),
+            // 'accent_color' => new StringValue(['value' => '#ffff00']),
+            // 'allow_flexible_color' => new BoolValue(['value' => false]),
+            // Whitelisted accounts only: Sets the format setting that the ad will be served in.
+            // 'format_setting' => DisplayAdFormatSetting::NON_NATIVE
+        ]);
+
+        // Creates a new ad group ad.
+        $adGroupAd = new AdGroupAd([
+            'ad' => new Ad([
+                'responsive_display_ad' => $responsiveDisplayAdInfo,
+                'final_urls' => [new StringValue(['value' => 'http://www.example.com/'])]
+            ]),
+            'ad_group' => new StringValue(['value' => $adGroupResourceName])
+        ]);
+
+        // Creates an ad group ad operation.
+        $adGroupAdOperation = new AdGroupAdOperation();
+        $adGroupAdOperation->setCreate($adGroupAd);
+
+        // Issues a mutate request to add the ad group ad.
+        $adGroupAdServiceClient = $googleAdsClient->getAdGroupAdServiceClient();
+        $response = $adGroupAdServiceClient->mutateAdGroupAds($customerId, [$adGroupAdOperation]);
+
+        /** @var AdGroupAd $addedAdGroupAd */
+        $addedAdGroupAd = $response->getResults()[0];
+        printf(
+            "Created ad group ad with resource name '%s'.%s",
+            $addedAdGroupAd->getResourceName(),
+            PHP_EOL
+        );
+    }
+
+    /**
+     * Adds an image asset to the Google Ads account.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @param string $imageUrl the image URL
+     * @param string $assetName the asset name
+     */
+    private static function uploadAsset(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $imageUrl,
+        string $assetName
+    ) {
+        // Creates an image content.
+        $imageContent = file_get_contents($imageUrl);
```

Comment:
This can be inlined below, especially when you have a comment explaining already.